setup.py

```diff
@@ -46,13 +46,9 @@
         "License :: OSI Approved :: MIT License",
         "Natural Language :: English",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
     ],
     description="Stransverse mass computation as a numpy ufunc.",
-    install_requires=["numpy"],
+    install_requires=["numpy>=1.26"],
```

Comment:
Looking at these docs, they suggest `1.23.5` should be supportable at least:

https://numpy.org/doc/stable/dev/depending_on_numpy.html#numpy-2-0-specific-advice

Since it's just a lower bound, it should be fine that on e.g. python 3.12 this bound will be superseded.

Reply:
Sure - lemme do that and integrate that sort of info into the tests.

Reply:
Excellent, thanks!

Somewhat related; I think we need to force the build to happen with numpy2. Example uses the `[build-system]` section of pyproject.toml. Not sure what the old-style incantation is.

Migrating to `pyproject.toml` is on the bucket list, but hopefully not required to do right now.

Reply:
Nah - migrating to `pyproject.toml` is not necessary (but would be easy and make some of your deployment stuff much more clean). 

A clearly useful migration would be to use pybind11 instead of doing it all yourself!

Reply:
> A clearly useful migration would be to use pybind11 instead of doing it all yourself!

I was & am keen to keep the dependency set as light as possible for this package. It's very simple, and no new features are planned. (Was a conscious decision to avoid pybind or other wrapper-layers when writing originally!)

Reply:
It's header only C++, and manages the interface to numpy in a way that maintains all the backwards compatibility. So it really is minimal and reduces maintenance work. 

Numpy 2 compatibility would have come for free just switching to 2.12. Much less pain in the long run (at least with the package's maturity as it is now).

Reply:
Have you used pybind11 to create ufuncs before? Thinking back I seem to recall that it wasn't supported. Actually not sure that it is now from a skim of the source code?

Anyway, even if it were possible I've actually been pleasantly surprised that no code changes have been required to get it building! Seems to be isolated to fiddly build system stuff.

Reply:
That was indeed lucky!